src/Commands/Site/GetSite.cs

```diff
@@ -10,13 +10,49 @@ namespace PnP.PowerShell.Commands.Site
     [OutputType(typeof(Microsoft.SharePoint.Client.Site))]
     public class GetSite : PnPRetrievalsCmdlet<Microsoft.SharePoint.Client.Site>
     {
+        [Parameter(Mandatory = false)]
+        public SwitchParameter GetVersionPolicy;
+
         protected override void ExecuteCmdlet()
         {
             DefaultRetrievalExpressions = new Expression<Func<Microsoft.SharePoint.Client.Site, object>>[] { s => s.Url, s => s.CompatibilityLevel };
             var site = ClientContext.Site;
-            ClientContext.Load(site, RetrievalExpressions);
-            ClientContext.ExecuteQueryRetry();
-            WriteObject(site);
+
+            if (ParameterSpecified(nameof(GetVersionPolicy)))
+            {
+                site.EnsureProperties(s => s.Url, s => s.VersionPolicyForNewLibrariesTemplate, s => s.VersionPolicyForNewLibrariesTemplate.VersionPolicies);
+
+                var s = new PSObject();
```

Comment:
Could it be an idea to create a model for this instead of using a freeform PSObject? This is how we do it for the other cmdlets as well.

Reply:
@KoenZomers Thanks so much for the suggestion! Added a new class for returning version policy object. Could you please check again? The tests results in the PR description is also updated.